hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,373 @@
+
+    enable_interface = None
+    add_vlan = None
+    remove_vlan = None
+    set_native_vlan = None
+    remove_native_vlan = None
+    remove_default_vlan = None
+    set_trunk_mode = None
+
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'username': basestring,
+            'hostname': basestring,
+            'password': basestring,
+        }).validate(kwargs)
+
+    def session(self):
+        return self
+
+    def disconnect(self):
+        pass
+
+    def _create_session(self):
+        dev = Device(
+                host=self.hostname, user=self.username, passwd=self.password
+                )
+        dev.bind(cfg=Config)
+        return dev
+
+    def _set_commands(self, port, vlan_id):
+        """ Set of common `set` commands used by other functions."""
+        port_set = "set interfaces {port}".format(port=port)
+        port_del = "delete interfaces {port}".format(port=port)
+
+        self.enable_interface = """
+        {port_del} disable""".format(port_del=port_del)
+
+        self.set_trunk_mode = """
+        {port_set} unit 0 family ethernet-switching interface-mode trunk
+        """.format(port_set=port_set)
+
+        self.add_vlan = """
+        {port_set} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port_set=port_set, vlan_id=vlan_id)
+
+        self.remove_vlan = """
+            {port_del} unit 0 family ethernet-switching vlan members {vlan_id}
+            """.format(port_del=port_del, vlan_id=vlan_id)
+
+        self.set_native_vlan = """
+        {port_set} native-vlan-id {vlan_id}
+        {port_set} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port_set=port_set, vlan_id=vlan_id)
+
+        self.remove_native_vlan = """
+        {port_del} native-vlan-id
+        {port_del} unit 0 family ethernet-switching vlan members {vlan_id}
+        """.format(port_del=port_del, vlan_id=vlan_id)
+
+        self.remove_default_vlan = """
+        {port_del} unit 0 family ethernet-switching vlan members default
+        """.format(port_del=port_del)
+
+    def _clear_set_commands(self):
+        """ Clears the values of the following variables."""
+        self.enable_interface = None
+        self.set_trunk_mode = None
+        self.add_vlan = None
+        self.remove_vlan = None
+        self.set_native_vlan = None
+        self.remove_native_vlan = None
+        self.remove_default_vlan = None
+
+    def _set_load_config(self, o_session, set_command):
+        """ Load configuration changes using the 'set' method.
+        `o_session`: session object with connection to switch already open.
+        `set_command`: Multi-line `set` method based command string.
+        """
+        jun = o_session
+        try:
+            jun.cfg.load(set_command, format="set")
+        except ConfigLoadError as e:
+            if e.message == "warning: statement not found":
+                pass
```

Comment:
sorry but it's' not clear what's happening here. can you add a comment here?

 If there's an error message then pass and then what? do you want to write it to logs or something?

Reply:
ok, I will add an comment there. sometimes the element on which the action is to be taken might not be present there. In that case this warning is  raised. It is harmless. To bring some sense of idempotency to the operations I implemented this. eg. to enable a port you have to delete the `disable` tag, if you do it again there is no `disable` tag for deletion, then the session will raise this warning.